test/functional-tests/index.js

```diff
@@ -383,4 +391,118 @@ describe('tests', () => {
         }));
     });
   });
+
+  describe('#Postcode-Lookup', () => {
+    describe('default postcode lookup behaviour', () => {
+      before(() => {
+        app = App(require('./apps/postcode-lookup-default')({ port })).listen(port);
+        port = app.address().port;
+      });
+
+      after(() => {
+        app.close();
+      });
+
+      it('stays on the postcode entry page if the postcode input is left as blank', () => browser.url('/postcode-default-one')
+        .$('input')
+        .setValue('')
+        .submitForm('form')
+        .getUrl()
+        .then(url => {
+          assert.ok(url.includes('postcode-default-one'));
+        }));
+
+      it('stays on the postcode entry page if the entered postcode is of an invalid format', () => browser.url('/postcode-default-one')
+        .$('input')
+        .setValue('abc123')
+        .submitForm('form')
+        .getUrl()
+        .then(url => {
+          assert.ok(url.includes('postcode-default-one'));
+        }));
+
+      it('redirects to the manual step if the enter address manually link is clicked', () => browser.url('/postcode-default-one')
+        .$('a[href*="step=manual"]')
+        .click()
+        .getUrl()
+        .then(url => {
+          assert.ok(url.includes('step=manual'));
+        }));
+
+      it('redirects to the lookup step on a successful postcode lookup', () => browser.url('/postcode-default-one')
```

Comment:
Once you managed to mock the request, we need to test on success lookup, the correct results are displayed

Reply:
Added a unit test to cover this.